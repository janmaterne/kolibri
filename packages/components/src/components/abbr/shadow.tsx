import type { JSX } from '@stencil/core';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import type { AbbrAPI, AbbrStates, LabelPropType } from '../../schema';
import { validateLabel } from '../../schema';

/**
 * @slot - The abbreviation (short form).
 */
@Component({
	tag: 'kol-abbr',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAbbr implements AbbrAPI {
	public render(): JSX.Element {
		return (
			<Host class="kol-abbr">
				<abbr>
					<slot />
				</abbr>
				{this.state._label ? ` (${this.state._label})` : ''}
			</Host>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label?: LabelPropType;

	@State() public state: AbbrStates = {
		_label: '', // âš  required
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label);
	}
}
