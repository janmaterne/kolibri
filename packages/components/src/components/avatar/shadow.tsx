import type { JSX } from '@stencil/core';
import { Component, h, Prop } from '@stencil/core';

import type { AvatarProps } from '../../schema';
import { KolAvatarWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-avatar',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAvatar implements AvatarProps {
	public render(): JSX.Element {
		return <KolAvatarWcTag _src={this._src} _label={this._label}></KolAvatarWcTag>;
	}

	/**
	 * Sets the image `src` attribute to the given string.
	 */
	@Prop() public _src?: string;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: string;
}
