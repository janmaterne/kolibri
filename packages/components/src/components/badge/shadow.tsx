import type { BadgeAPI, BadgeStates, ButtonProps, KoliBriIconsProp, LabelPropType, PropColor, Stringified } from '../../schema';
import { featureHint, handleColorChange, objectObjectHandler, parseJson, setState, validateColor, validateIcons } from '../../schema';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import { KolSpanFc } from '@fc';

import { nonce } from '../../utils/dev.utils';

import type { JSX } from '@stencil/core';
import { KolButtonWcTag } from '../../core/component-names';
featureHint(`[KolBadge] Optimierung des _color-Properties (rgba, rgb, hex usw.).`);

@Component({
	tag: 'kol-badge',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolBadge implements BadgeAPI {
	private bgColorStr = '#000';
	private colorStr = '#fff';
	private readonly id = nonce();

	private renderSmartButton(props: ButtonProps): JSX.Element {
		return (
			<KolButtonWcTag
				_ariaControls={this.id}
				_customClass={props._customClass}
				_disabled={props._disabled}
				_hideLabel={true}
				_icons={props._icons}
				_id={props._id}
				_label={props._label}
				_on={props._on}
				_tooltipAlign={props._tooltipAlign}
				_variant={props._variant}
			></KolButtonWcTag>
		);
	}

	public render(): JSX.Element {
		const hasSmartButton = typeof this.state._smartButton === 'object' && this.state._smartButton !== null;

		return (
			<Host class="kol-badge">
				<span
					class={{
						'smart-button': typeof this.state._smartButton === 'object' && this.state._smartButton !== null,
					}}
					style={{
						backgroundColor: this.bgColorStr,
						color: this.colorStr,
					}}
				>
					<KolSpanFc id={hasSmartButton ? this.id : undefined} allowMarkdown icons={this.state._icons} label={this._label} />
					{hasSmartButton && this.renderSmartButton(this.state._smartButton as ButtonProps)}
				</span>
			</Host>
		);
	}

	/**
	 * Defines the backgroundColor and foregroundColor.
	 */
	@Prop() public _color?: Stringified<PropColor> = '#000';

	/**
	 * Defines the icon classnames (e.g. `_icons="fa-solid fa-user"`).
	 */
	@Prop() public _icons?: Stringified<KoliBriIconsProp>;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Allows to add a button with an arbitrary action within the element (_hide-label only).
	 */
	@Prop() public _smartButton?: Stringified<ButtonProps>;

	@State() public state: BadgeStates = {
		_color: {
			backgroundColor: '#000',
			foregroundColor: '#fff',
		},
		_icons: {},
	};

	private handleColorChange = (value: unknown) => {
		const colorPair = handleColorChange(value);
		this.bgColorStr = colorPair.backgroundColor;
		this.colorStr = colorPair.foregroundColor as string;
	};

	@Watch('_icons')
	public validateIcons(value?: KoliBriIconsProp): void {
		validateIcons(this, value);
	}

	@Watch('_color')
	public validateColor(value?: Stringified<PropColor>): void {
		validateColor(this, value, {
			defaultValue: '#000',
			hooks: {
				beforePatch: this.handleColorChange,
			},
		});
	}

	@Watch('_smartButton')
	public validateSmartButton(value?: ButtonProps | string): void {
		objectObjectHandler(value, () => {
			try {
				value = parseJson<ButtonProps>(value as string);
				// eslint-disable-next-line no-empty
			} catch (e) {
				// value behält den ursprünglichen Wert
			}
			setState(this, '_smartButton', value);
		});
	}

	public componentWillLoad(): void {
		this.validateIcons(this._icons);
		this.validateColor(this._color);
		this.validateSmartButton(this._smartButton);
	}
}
