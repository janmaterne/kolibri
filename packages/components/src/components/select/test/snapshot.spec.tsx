import { KolSelectTag } from '@component-names';
import type { SelectProps } from '@schema';
import { executeInputSnapshotTests } from '@testing';

import { KolSelect } from '../shadow';

const options = [
	{
		label: 'Frau',
		value: 'Frau',
		disabled: true,
	},
	{
		label: 'Herr',
		value: 'Herr',
	},
	{
		label: 'Divers',
		value: 'Divers',
	},
];

executeInputSnapshotTests<SelectProps>(KolSelectTag, [KolSelect], {
	_options: options,
});

executeInputSnapshotTests<SelectProps>(KolSelectTag, [KolSelect], {
	_options: options,
	_multiple: true,
});
