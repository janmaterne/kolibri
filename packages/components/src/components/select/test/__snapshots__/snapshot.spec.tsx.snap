// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _accessKey="V" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select accesskey="V" autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _alert=true 1`] = `
<kol-select _alert="" _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _disabled=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" disabled="" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _hint="Hint" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select aria-describedby="id-nonce-hint" autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-select _hide-error="" _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _accessKey="V" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select accesskey="V" autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _alert=true 1`] = `
<kol-select _alert="" _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _disabled=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" disabled="" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _hint="Hint" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select aria-describedby="id-nonce-hint" autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-select _hide-error="" _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _readOnly=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" _readonly="" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _shortKey="S" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _shortkey="S" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          S
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true _touched=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" _touched="" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _multiple=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" multiple="" name="field" spellcheck="false">
            <option disabled="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _readOnly=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" _readonly="" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _shortKey="S" 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _shortkey="S" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          S
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] _touched=true 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" _touched="" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _alert="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;

exports[`kol-select should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _options=[{"label":"Frau","value":"Frau","disabled":true},{"label":"Herr","value":"Herr"},{"label":"Divers","value":"Divers"}] 1`] = `
<kol-select _placeholder="Hier steht ein Platzhaltertext" class="has-value kol-select">
  <template shadowrootmode="open">
    <kol-input _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="select" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <form>
          <input hidden="" type="submit">
          <select autocapitalize="off" autocorrect="off" id="id-nonce" name="field" spellcheck="false">
            <option disabled="" selected="" value="-0">
              Frau
            </option>
            <option value="-1">
              Herr
            </option>
            <option value="-2">
              Divers
            </option>
          </select>
        </form>
      </div>
    </kol-input>
  </template>
</kol-select>
`;
